inotify/adapters.py

```diff
@@ -140,12 +142,14 @@ def _get_event_names(self, event_type):
 
         return names
 
-    def _handle_inotify_event(self, wd, event_type):
+    def _handle_inotify_event(self, inotify_fd, event_type):
         """Handle a series of events coming-in from inotify."""
 
-        names = self._get_event_names(event_type)
+        # Set the inotify file-descriptor to non-blocking since blocks have been seen on os.read
+        flag = fcntl.fcntl(inotify_fd, fcntl.F_GETFL)
+        fcntl.fcntl(inotify_fd, fcntl.F_SETFL, flag | os.O_NONBLOCK)
```

Comment:
Can we move this to `__init__`? Do we need to call this every time we get an event?

Reply:
I'm not sure. Based on the above it should at least be run on each watch descriptor. I'm pretty paranoid about not doing it, and I guess the cost is not high?